contracts/token/ERC677.cairo

```diff
@@ -0,0 +1,373 @@
+%lang starknet
+%builtins pedersen range_check ecdsa
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.math import assert_not_zero
+from starkware.cairo.common.uint256 import (
+    Uint256, uint256_add, uint256_sub, uint256_le, uint256_lt, uint256_check
+)
+
+#
```

Comment:
The `IERC677Receiver` interface so that `transferAndCall` can pass data to the receiver.